docs/legacy/storage-management.asciidoc

```diff
@@ -248,6 +248,38 @@ POST /apm-*/_delete_by_query
 
 See {ref}/docs-delete-by-query.html[delete by query] for further information on this topic.
 
+[float]
+[[delete-other-apm-data]]
+===== Delete other APM data
+
+In addition to trace event data, there are other APM indices that you may wish to delete data from.
+
+APM {kibana-ref}/custom-links.html[custom links] are stored in the `.apm-custom-link` index.
+To delete all custom links, run:
+
+[source,console]
+----
+POST /.apm-custom-link/_delete_by_query
+{
+  "query": {
+    "match_all": {}
+  }
+}
+----
+
+APM agent {kibana-ref}/agent-configuration.html[central configuration] data is stored in the `.apm-agent-configuration` index.
+To delete all central agent configurations, run:
+
+[source,console]
+----
+POST /.apm-agent-configuration/_delete_by_query
+{
+  "query": {
+    "match_all": {}
+  }
+}
```

Comment:
Agent configurations are sync'd with fleet so deleting from here might not be enough. I'll have to go check when and how the syncing mechanism works.

Reply:
Upon creating/deleting an agent config a call is made to sync the state to fleet: 
https://github.com/elastic/kibana/blob/7591fb61556bc56aaae725d6db51bd80e958d91b/x-pack/plugins/apm/server/routes/settings/agent_configuration/route.ts#L115-L125

If the user simply deletes the agent config index the sync will not happen until they manually create a new agent config. This means that agents will still receive the old agent configs (via fleet), even though the agent config index was created. I can see how this can be confusing.

I'm not sure what the best solution would be. We could sync the agent configs periodically (eg. every hour), or when Kibana starts.

Reply:
Ahh. Interesting. I suppose we could document that the user needs to create a temporary agent config value to sync the index. I assume deleting the entire index is only useful for users with a large number of configuration settings, so deleting the index and creating a single config could be faster than clicking all of the `trash` icons?

Reply:
Thanks! Fixed in [`102ab5a`](https://github.com/elastic/apm-server/pull/8145/commits/102ab5aa2a3cd29482118c9e16f19cd3f720cf52). Appreciate the help; it's good to have you back.